### PR TITLE
ISPN-15348 Static analysis

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-15348

### CODE REVIEW

**Comment by ryanemerson** [comment]
```java
[2023-11-29T16:02:42.299Z] 11:02:42.029 [INFO] -------------------------------------------------------------
[2023-11-29T16:02:42.299Z] 11:02:42.029 [INFO] -------------------------------------------------------------
[2023-11-29T16:02:42.299Z] 11:02:42.029 [ERROR] COMPILATION ERROR : 
[2023-11-29T16:02:42.299Z] 11:02:42.030 [INFO] -------------------------------------------------------------
[2023-11-29T16:02:42.299Z] 11:02:42.030 [ERROR] /home/infinispan/workspace/Infinispan_PR-11532/jcache/embedded/src/test/java/org/infinispan/jcache/JCacheLoaderTest.java:[100,67] error: cannot find symbol
[2023-11-29T16:02:42.299Z]   symbol:   method intValue()
[2023-11-29T16:02:42.299Z]   location: class Object
[2023-11-29T16:02:42.299Z] 11:02:42.030 [ERROR] /home/infinispan/workspace/Infinispan_PR-11532/jcache/embedded/src/test/java/org/infinispan/jcache/JCacheLoaderTest.java:[110,71] error: cannot find symbol
[2023-11-29T16:02:42.299Z]   symbol:   method intValue()
[2023-11-29T16:02:42.299Z]   location: class Object
```

**Comment by tristantarrant** [comment]
Yes, I'll fix it once https://github.com/infinispan/infinispan/pull/11522 is in

**Comment by tristantarrant** [comment]
This is now ready

File: commons/all/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSet.java [hunk 1]
```diff
@@ -59,13 +60,14 @@ public AttributeSet(Class<?> klass, String name, AttributeSet attributeSet, Attr
    public AttributeSet(Class<?> klass, String name, AttributeSet attributeSet, AttributeDefinition<?>[] attributeDefinitions, RemovedAttribute[] removedAttributes) {
       this.klass = klass;
       this.name = name;
+      Map<String, Attribute<?>> attributes;
       if (attributeSet != null) {
-         this.attributes = new LinkedHashMap<>(attributeDefinitions.length + attributeSet.attributes.size());
```
Comment (wburns) [inline commons/all/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSet.java:63]: Are we sure we want to change this to be a HashMap? All of the methods including toString would probably be okay but the write method for the xml writing would change the ordering of the elements written which could be a maintenance pain. WDYT?

Comment (wburns) [reply commons/all/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSet.java:63]: Is this change why https://github.com/infinispan/infinispan/pull/11532/commits/45b1af0bfeef39f75bf7f8d15fc027be2e36d7c7 is needed?

**Comment by wburns** [comment]
Can we separate https://github.com/infinispan/infinispan/pull/11532/commits/b2db606d54977227dbe8799edb6bfde437c040c4 regarding the command modules to a different PR? I would like to look at this separately and it isn't as trivial as many of the other changes.

**Comment by tristantarrant** [comment]
I have removed the `ModuleProperties` changes, to be issued in a separate PR. I have also reintroduced the `LinkedHashMap` to `AttributeSet` while keeping the parsing/serialization order changes.

**Pull request merged**